daceml/onnx/op_implementations/img_op_implementations.py

```diff
@@ -0,0 +1,360 @@
+                                                code="output = image_in")
+
+        x_idx = _2d_sliding_window_index_expr(x_or_y="x",
+                                              stride=stride_x,
+                                              kernel_size=filter_hx)
+        y_idx = _2d_sliding_window_index_expr(x_or_y="y",
+                                              stride=stride_y,
+                                              kernel_size=filter_hy)
+
+        image_memlet = dace.Memlet("X[b, c, {}, {}]".format(x_idx, y_idx))
+
+        new_state.add_edge(inner_me, None, compute_tasklet, "image_in",
+                           image_memlet)
+
+        # hook up X
+        read_X = new_state.add_read("X")
+        inner_image_memlet = propagation.propagate_memlet(
+            new_state, image_memlet, inner_me, False)
+        outer_image_memlet = propagation.propagate_memlet(
+            new_state, inner_image_memlet, outer_me, False)
+        new_state.add_edge(outer_me, None, inner_me, None, inner_image_memlet)
+        new_state.add_edge(read_X, None, outer_me, None, outer_image_memlet)
+
+        # hook up outputs
+        output_memlet = dace.Memlet("Y[b, c, out_x, out_y]",
+                                    wcr="lambda x, y: max(x, y)")
+        inner_output_memlet = propagation.propagate_memlet(
+            new_state, output_memlet, inner_me, False)
+        outer_output_memlet = propagation.propagate_memlet(
+            new_state, inner_output_memlet, outer_me, False)
+        new_state.add_edge(compute_tasklet, "output", inner_mx, None,
+                           output_memlet)
+
+        write_Y = new_state.add_write("Y")
+        new_state.add_edge_pair(outer_mx, inner_mx, write_Y,
+                                inner_output_memlet, outer_output_memlet)
+
+        new_sdfg.fill_scope_connectors()
+        return new_sdfg
+
+
+@op_implementation(op="Conv", name="pure")
+class PureConv2D(ONNXForward):
+    """ The "trivial" convolution implementation, i.e. two nested maps.
+    """
+    @staticmethod
+    def forward_can_be_applied(node: ONNXOp, state: SDFGState,
+                               sdfg: SDFG) -> bool:
+        X = in_desc_with_name(node, state, sdfg, "X")
+        W = in_desc_with_name(node, state, sdfg, "W")
+        try:
+            B = in_desc_with_name(node, state, sdfg, "B")
+        except Exception as e:
+            B = None
+
+        image_dims = len(X.shape) - 2
+        num_filters = W.shape[0]
+        num_channels = X.shape[1]
+
+        if (X.dtype not in [dace.float16, dace.float32, dace.float64]
+                or W.dtype not in [dace.float16, dace.float32, dace.float64]):
+            return False
+
+        # only do 2D for now
+        if len(X.shape) != 4 or len(W.shape) != 4:
+            return False
+
+        if node.group != 1:
+            return False
+
+        if num_channels != W.shape[1]:
+            return False
+
+        if node.dilations is not None and (not all(d == 1
```

Comment:
do we need those for any of the current use cases?

Reply:
We'll add them as we need them